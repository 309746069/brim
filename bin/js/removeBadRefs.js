/* @flow */
const child_process = require("child_process")

function removeBadRefs() {
  const paths = []
  const files = child_process.execSync("./bin/badrefs").toString()

  files.split("\n").forEach((file) => {
    const [path, line, endline] = file.split(":")
    if (path && line && endline) {
      paths.push(path)
      removeLine(path, line, endline)
    }
  })

  return paths
}

function removeLine(path, line, endline) {
  let cmd = `sed -i "" '${line},${endline}s/.*//' ${path}`
  child_process.execSync(cmd)
}

function run() {
  console.log("Removing bad refs")
  removeBadRefs()
  console.log("Done")
}

run()
