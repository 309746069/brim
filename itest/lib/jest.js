/* @flow */

import {Application} from "spectron"

import {selectors} from "../../src/js/test/integration"

export const TestTimeout = 60000

// This is trying to make Flow happy for passing in the done function from
// Jest.
export type Done = {
  (string | Error | void): void,
  fail?: (string | Error | void) => void
}

export const handleError = async (
  app: Application,
  initialError: Error,
  done: Done
) => {
  let realError = undefined
  let notificationError = undefined
  console.log(`handleError: Test hit exception: ${initialError.message}`)
  console.log("handleError: Looking for any desktop app notifications")
  try {
    notificationError = await app.client.getHTML(selectors.notification, false)
  } catch (e) {
    notificationError = undefined
  }
  if (notificationError) {
    realError = new Error(
      "App notification '" +
        notificationError +
        "' (initial error: '" +
        initialError +
        "'"
    )
  } else {
    console.log("handleError: desktop app notification not found")
    realError = initialError
  }
  done.fail && done.fail(realError)
}
