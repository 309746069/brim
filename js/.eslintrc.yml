parserOptions:
  ecmaVersion: 6
  ecmaFeatures:
    jsx: true
    experimentalObjectRestSpread: true
  sourceType: "module"

env:
  browser: true
  jest: true
  es6: true

extends:
  - eslint:recommended
  - plugin:react/recommended

rules:
  no-unused-vars:
    - error
    - argsIgnorePattern: "^_"
      varsIgnorePattern: "^_"
  semi:
    - error
    - never
  no-console:
    - off
  react/prop-types:
    - ignore
  quotes:
    - error
    - double
  react/no-find-dom-node:
    - ignore
  no-debugger:
    - off
