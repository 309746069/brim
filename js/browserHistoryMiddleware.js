export default _store => next => {
  window.onpopstate = _event => pop()

  return action => {
    if (action.type === "MAIN_SEARCH_REQUEST" && action.saveToHistory) {
      push()
    }
    next(action)
  }
}

function push() {}
function pop() {}
