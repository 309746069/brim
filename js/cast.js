import moment from "moment"

const BRO_TS_FORMAT = "X.SSSSSS"

export function toDate(string) {
  return moment.utc(string, BRO_TS_FORMAT).toDate()
}

export function toInt(string) {
  return parseInt(string)
}

export function toTs(date) {
  return moment.utc(date).format(BRO_TS_FORMAT)
}

export function toSec(date) {
  return moment.utc(date).unix()
}

export const toMoment = ({sec, ns}) => {
  return moment.utc(sec + "." + ns, BRO_TS_FORMAT)
}

export const fromNanoTsToDate = ts => new Date(ts / 1000000.0)

export const fromNanoTsToMoment = ts => moment.utc(fromNanoTsToDate(ts))
