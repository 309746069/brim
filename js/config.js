export default {
  FULL_MOMENT_FORMAT: "MMM DD, YYYY HH:mm:ss.SSS",
  TIME_MOMENT_FORMAT: "HH:mm:ss.SSS"
}
