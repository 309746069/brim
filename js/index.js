import React from "react"
import ReactDOM from "react-dom"
import {AppContainer} from "react-hot-loader"
import {Provider} from "react-redux"
import initStore from "./initStore"
import {HashRouter} from "react-router-dom"
import XApp from "./connectors/XApp"
const store = initStore()

ReactDOM.render(
  <AppContainer>
    <Provider store={store}>
      <HashRouter>
        <XApp />
      </HashRouter>
    </Provider>
  </AppContainer>,
  document.getElementById("app-root")
)
