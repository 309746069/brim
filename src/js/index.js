import React from "react"
import ReactDOM from "react-dom"
import {Provider} from "react-redux"
import initStore from "./initStore"
import {HashRouter} from "react-router-dom"
import XApp from "./connectors/XApp"
const store = initStore()

ReactDOM.render(
  <Provider store={store}>
    <HashRouter>
      <XApp />
    </HashRouter>
  </Provider>,
  document.getElementById("app-root")
)
