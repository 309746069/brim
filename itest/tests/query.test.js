/* @flow */

// The purpose of this file is to demonstrate that basic Spectron interaction
// can work in a CI environment. The tests don't claim to be meaningful other
// than showing Spectron works in a headless environment.
//
// The setup/teardown was taken from
// https://github.com/electron/spectron/#usage

import {
  getSearchSpeed,
  getSearchTime,
  logIn,
  searchDisplay,
  setSpan,
  startSearch,
  waitForLoginAvailable,
  waitForHistogram,
  waitForSearch,
  writeSearch
} from "../lib/app.js"
import {TestTimeout, handleError} from "../lib/jest.js"

const Application = require("spectron").Application
const electronPath = require("electron") // Require Electron from the binaries included in node_modules.
const path = require("path")

describe("Query tests", () => {
  let app
  beforeEach(() => {
    // TODO: Move this logic into a library, especially as it expands.
    app = new Application({
      path: electronPath,
      args: [path.join(__dirname, "..", "..")]
    })
    return app.start().then(() => app.webContents.send("resetState"))
  })

  afterEach(() => {
    if (app && app.isRunning()) {
      return app.stop()
    }
  })

  test(
    "query path=weird | sort",
    (done) => {
      waitForLoginAvailable(app)
        .then(() => logIn(app))
        .then(() => waitForHistogram(app))
        .then(() => waitForSearch(app))
        .then(() => writeSearch(app, "_path=weird | sort"))
        .then(() => startSearch(app))
        .then(() => waitForSearch(app))
        .then(() => searchDisplay(app))
        .then((results) => {
          expect(results).toMatchSnapshot()
        })
        .then(() => getSearchSpeed(app))
        .then((searchSpeed) => {
          expect(searchSpeed).toBeGreaterThan(0)
          expect(searchSpeed).toBeLessThan(1000)
        })
        .then(() => getSearchTime(app))
        .then((searchTime) => {
          expect(searchTime).toBeGreaterThan(0)
          expect(searchTime).toBeLessThan(5)
          done()
        })
        .catch((err) => {
          handleError(app, err, done)
        })
    },
    TestTimeout
  )

  test(
    "query path=_http | count()",
    (done) => {
      waitForLoginAvailable(app)
        .then(() => logIn(app))
        .then(() => waitForHistogram(app))
        .then(() => waitForSearch(app))
        .then(() => writeSearch(app, "_path=http | count()"))
        .then(() => startSearch(app))
        .then(() => waitForSearch(app))
        .then(() => searchDisplay(app))
        .then((results) => {
          expect(results).toMatchSnapshot()
        })
        .then(() => getSearchSpeed(app))
        .then((searchSpeed) => {
          expect(searchSpeed).toBeGreaterThan(0)
          expect(searchSpeed).toBeLessThan(1000)
        })
        .then(() => getSearchTime(app))
        .then((searchTime) => {
          expect(searchTime).toBeGreaterThan(0)
          expect(searchTime).toBeLessThan(5)
          done()
        })
        .catch((err) => {
          handleError(app, err, done)
        })
    },
    TestTimeout
  )

  test(
    "query _path=http | count() by id.resp_p | sort",
    (done) => {
      waitForLoginAvailable(app)
        .then(() => logIn(app))
        .then(() => waitForHistogram(app))
        .then(() => waitForSearch(app))
        .then(() =>
          writeSearch(app, "_path=http | count() by id.resp_p | sort")
        )
        .then(() => startSearch(app))
        .then(() => waitForSearch(app))
        .then(() => searchDisplay(app))
        .then((results) => {
          expect(results).toMatchSnapshot()
        })
        .then(() => getSearchSpeed(app))
        .then((searchSpeed) => {
          expect(searchSpeed).toBeGreaterThan(0)
          expect(searchSpeed).toBeLessThan(1000)
        })
        .then(() => getSearchTime(app))
        .then((searchTime) => {
          expect(searchTime).toBeGreaterThan(0)
          expect(searchTime).toBeLessThan(5)
          done()
        })
        .catch((err) => {
          handleError(app, err, done)
        })
    },
    TestTimeout
  )

  test(
    "query _path=http | every 5m count()",
    (done) => {
      waitForLoginAvailable(app)
        .then(() => logIn(app))
        .then(() => waitForHistogram(app))
        .then(() => waitForSearch(app))
        .then(() => writeSearch(app, "_path=http | every 5m count()"))
        .then(() => startSearch(app))
        .then(() => waitForSearch(app))
        .then(() => searchDisplay(app))
        .then((results) => {
          expect(results).toMatchSnapshot()
        })
        .then(() => getSearchSpeed(app))
        .then((searchSpeed) => {
          expect(searchSpeed).toBeGreaterThan(0)
          expect(searchSpeed).toBeLessThan(1000)
        })
        .then(() => getSearchTime(app))
        .then((searchTime) => {
          expect(searchTime).toBeGreaterThan(0)
          expect(searchTime).toBeLessThan(5)
          done()
        })
        .catch((err) => {
          handleError(app, err, done)
        })
    },
    TestTimeout
  )

  test(
    "query * | count(); switch to whole space",
    (done) => {
      waitForLoginAvailable(app)
        .then(() => logIn(app))
        .then(() => waitForHistogram(app))
        .then(() => waitForSearch(app))
        .then(() => writeSearch(app, "* | count()"))
        .then(() => startSearch(app))
        .then(() => waitForSearch(app))
        .then(() => searchDisplay(app))
        .then((results) => {
          expect(results).toMatchSnapshot()
        })
        .then(() => setSpan(app, "Whole Space"))
        .then(() => waitForSearch(app))
        .then(() => searchDisplay(app))
        .then((results) => {
          expect(results).toMatchSnapshot()
        })
        .then(() => getSearchSpeed(app))
        .then((searchSpeed) => {
          expect(searchSpeed).toBeGreaterThan(0)
          expect(searchSpeed).toBeLessThan(1000)
        })
        .then(() => getSearchTime(app))
        .then((searchTime) => {
          expect(searchTime).toBeGreaterThan(0)
          expect(searchTime).toBeLessThan(5)
          done()
        })
        .catch((err) => {
          handleError(app, err, done)
        })
    },
    TestTimeout
  )
})
