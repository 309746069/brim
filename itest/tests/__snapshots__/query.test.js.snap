// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Query tests query "* | count()"; switch to whole space 1`] = `
Array [
  "count",
  "959",
]
`;

exports[`Query tests query "* | count()"; switch to whole space 2`] = `
Array [
  "count",
  "1,031,683",
]
`;

exports[`Query tests query "_path=http | count() by id.resp_p | sort" 1`] = `
Array [
  "id.resp_p",
  "count",
  "8080",
  "6",
  "80",
  "55",
]
`;

exports[`Query tests query "_path=http | count()" 1`] = `
Array [
  "count",
  "61",
]
`;

exports[`Query tests query "_path=http | every 5m count()" 1`] = `
Array [
  "ts",
  "count",
  "04/13/15",
  "09:30",
  "00.0000",
  "55",
  "04/13/15",
  "09:25",
  "00.0000",
  "2",
  "04/13/15",
  "09:10",
  "00.0000",
  "2",
  "04/13/15",
  "09:05",
  "00.0000",
  "2",
]
`;

exports[`Query tests query "_path=weird | sort" 1`] = `
Array [
  "ts",
  "_path",
  "uid",
  "id.orig_h",
  "id.orig_p",
  "id.resp_h",
  "id.resp_p",
  "name",
  "addl",
  "notice",
  "peer",
  "04/13/15",
  "09:31",
  "27.8210",
  "weird",
  "CbtgYm2YYGmfYsyaC4",
  "192.168.0.54",
  "53147",
  "216.58.209.134",
  "80",
  "inflate_failed",
  "-",
  "F",
  "bro",
  "04/13/15",
  "09:31",
  "51.9870",
  "weird",
  "-",
  "-",
  "-",
  "-",
  "-",
  "unknown_protocol_2",
  "-",
  "F",
  "bro",
  "04/13/15",
  "09:32",
  "45.1220",
  "weird",
  "Ctd0Gv1XC3Z0UV1iX",
  "192.168.0.53",
  "3682",
  "192.168.0.1",
  "8080",
  "unknown_HTTP_method",
  "UNSUBSCRIBE",
  "F",
  "bro",
  "04/13/15",
  "09:32",
  "45.1400",
  "weird",
  "CZehjU3zE0mxEXiC41",
  "192.168.0.53",
  "3683",
  "192.168.0.1",
  "8080",
  "unknown_HTTP_method",
  "UNSUBSCRIBE",
  "F",
  "bro",
]
`;
